telegram/ext/filters.py

```diff
@@ -65,6 +68,15 @@ def __or__(self, other):
     def __invert__(self):
         return InvertedFilter(self)
 
+    def __str__(self):
+        # Do not rely on classes overwriting __init__ to set a name
+        # so we can keep backwards compatibility
+        if not hasattr(self, 'name') or self.name is None:
+            self.name = self.__class__.__name__
+        return self.name
+
+    __repr__ = __str__
```

Comment:
it's sufficient to implement `__repr__`.  It will implicitly get `__str__` working as you want